File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        set<int> unique{nums.begin(), nums.end()};
        int i = 0;
        for (int val : unique) {
            nums[i++] = val;
        }
        return i;
    }
};

File: Data Structures & Algorithms/single-number/submission-1.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        # brute force (O(n) time and space)
        # freq = {}
        # for num in nums:
        #     if num in freq:
        #         freq[num] = 2
        #     else: 
        #         freq[num] = 1
        
        # for k,v in freq.items():
        #     if v == 1: 
        #         return k
        
        # return -1

        # Take the bitwise op which finds the extra 1s
        # Since all appear twice, XOR leaves only the extra 1s
        res = 0
        for num in nums: 
            res ^= num

        return res


    